mailpoet/lib/Automation/Integrations/MailPoet/Subjects/SubscriberSubject.php

```diff
@@ -25,8 +25,16 @@ public function __construct(
     $this->subscribersRepository = $subscribersRepository;
 
     $this->fields = [
-      // email
-      new Field(
+      'id' => new Field(
```

Comment:
The ticket specs say to use the email field for the send welcome email action, but some of the things we need to do (checking that the email hasn't already been scheduled for them, checking that they are in fact subscribed to the segment) seems to require a subscriber object or ID. I could look up the subscriber with their email, but I don't know why that would be preferable to simply using the subscriber ID.

Once again I think I'm not understanding the general idea when it comes to `Field`s. I'm wondering why we wouldn't do something like this:

```php
$this->fields = [
  new Field(
    'mailpoet:subscriber',
    Field::TYPE_SUBSCRIBER,
    __('Subscriber', 'mailpoet'),
    function() {
      return $this->getSubscriber();
    }
  ),
```

and then have actions take care of things like retrieving an email address or subscription status?

Do they need to be something that can be easily represented by text, like in a UI, and we shouldn't have actions retrieving additional data at all?

I like the idea of returning the subscriber entity because otherwise we would need to keep adding fields to `SubscriberSubject` any time a new action requires new data and that could get pretty messy. I'm sure there's something major that I'm missing here though.

Reply:
> The ticket specs say to use the email field for the send welcome email action... to simply using the subscriber ID.

Good point, in that case we may add a field for the ID. I think we can't use the subscriber entity itself though. The reason there are the primitives such as `Field::TYPE_STRING` and `Field::TYPE_ENUM` is that in the UI, we'll display them as respective form fields (e.g. a select box) and allow also filtering (e.g. a field of type integer is greater than some value). Having that... maybe we could have a helper method on the subject to fetch the subscriber or something but maybe also the first step can be fetching it by ID in the action, if we need the full entity.

> Do they need to be something that can be easily represented by text, like in a UI, and we shouldn't have actions retrieving additional data at all?

Yep, that's what I meant above.

> I like the idea of returning the subscriber entity because otherwise we would need to keep adding fields to SubscriberSubject any time a new action requires new data and that could get pretty messy. I'm sure there's something major that I'm missing here though.

Would ID work? Or some way to get the subscriber in an action but keep the basic fields for UI, conditions, filters, etc.?

Reply:
I'm currently retrieving the subscriber and segment entities directly from the subjects, and I'm not sure I understand why it would be preferable to fetch them from an ID instead. I think I still need to wrap my head around exactly how the fields will work in the UI.

Reply:
I guess that falls into what I meant by `Having that... maybe we could have a helper method on the subject to fetch the subscriber or something`. Seems good to me.

The only thing I wanted to stress here is that for some non-MailPoet steps or actions (like conditions and stuff), we'll need to extract all the data we want to allow to be used and/or filtered to fields as conditions should know nothing about `SubscriberEntity`. On the other hand the MailPoet actions definitely can know about it and use it.